kerl

```diff
@@ -754,7 +668,131 @@ _flags() {
     esac
 }
 
+_dpkg() {
+    # gratefully stolen from
+    # https://superuser.com/questions/427318/test-if-a-package-is-installed-in-apt
+    # returns 0 (true) if found, 1 otherwise
+    echo "dpkg-query -Wf'${db:Status-abbrev}' \"$1\" 2>/dev/null | \grep -q '^i'"
+}
+
+_rpm() {
+    echo "rpm --quiet -q \"$1\" >/dev/null 2>&1"
+}
+
+_apk() {
+    echo "apk -e info \"$1\" >/dev/null 2>&1"
+}
+
+# To add package guessing magic for your Linux distro/package,
+# create a variable named _KPP_<distro>_pkgs where the content
+# is an array with packages, and then create a unique probe
+# command, in variable _KPP_<distro>_probe to check from the package manager.
+# It should return 0 if package installation is Ok, non-0 otherwise.
+
+_KPP_alpine_pkgs="openssl-dev make autoconf ncurses-dev gcc g++"
+_KPP_alpine_probe="_apk"
```

Comment:
If we decide to go by package manager we have less entries, but basically go back to where we were, without the possibility to define different package names per different distro.

Also, I tried a bunch of stuff here, mostly associative arrays (which could give us granularity such as "a command per package", but ultimately this is a POSIX-sh, so not much we can do there, with arrays that is)...

Reply:
Ran into this already, while testing `rhel`. Package for `g++` is `gcc-c++`. Don't know if it's the same for Fedora, for example...